libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetectorTest.java

```diff
@@ -89,6 +89,26 @@ public void isUserOffRoute_AssertTrueWhenTooFarFromStep() throws Exception {
     assertTrue(isUserOffRoute);
   }
 
+  @Test
+  public void isUserOffRoute_StepPointSize() throws Exception {
+    RouteProgress routeProgress = buildDefaultTestRouteProgress();
+    Point stepManeuverPoint = routeProgress.directionsRoute().legs().get(0).steps().get(0).maneuver().location();
+
+    Location firstUpdate = buildDefaultLocationUpdate(-77.0339782574523, 38.89993519985637);
+    offRouteDetector.isUserOffRoute(firstUpdate, routeProgress, options);
+
+    Point offRoutePoint = buildPointAwayFromPoint(stepManeuverPoint, 50, 90);
+    Location secondUpdate = buildDefaultLocationUpdate(offRoutePoint.longitude(), offRoutePoint.latitude());
+
+    // Manually remove all but one point to trigger the step point check
```

Comment:
What about extracting this block of code into a private method and give it a name based on the comment? This way the test will be easier to read and understand and comment will become superfluous so it won't be necessary.

Reply:
Sure, not a problem. Did that.